sunpy/lightcurve/sources/sit.py

```diff
@@ -0,0 +1,326 @@
+            elif delta == 'month':
+                for n in range( monthdelta(start_date, end_date) + 1):
+                    yield  add_months(start_date,n )
+            elif delta == 'year':  
+                for n in range(relativedelta(end_date, start_date).years +1):
+                    yield add_years(start_date,n)      
+
+
+
+        #creating a list of links to file for each date in TimeRange by looping and adding date of file to base url
+        url = []
+        url_meta = (start,end)
+        
+        
+        if type_of_average == '1hr' or type_of_average == '1day': 
+            for single_date in daterange(start, end, delta = 'year'):  
+                url.append(base_url + "{date:%Y}.txt".format(date=single_date))
+
+        elif type_of_average == '1min':
+            for single_date in daterange(start, end, delta = 'day'):
+                day_of_year = (str)(single_date.timetuple().tm_yday)
+                for i in [0,1,2]:
+                    if len(day_of_year) < 3:
+                        day_of_year = '0' + day_of_year
+                url.append(base_url + "{date:%Y}_".format(date = single_date) + day_of_year + ".txt".format(
+                    date=single_date))
+
+        elif type_of_average == '10min':
+            for single_date in daterange(start, end, delta = 'month'):
+                url.append(base_url + "{date:%Y}_{date:%m}.txt".format(
+                    date=single_date))
+
+
+        return url, url_meta
+
+    """
+    #Test Cases for above routine (will add separate test case in pytest for the lightcurve later when I complete it)
+    print _get_url_for_date_range('2009-10-02', '2009-10-04','ahead','1min','4He')
+    print _get_url_for_date_range('2008-06-02', '2010-02-04','behind','10min','Fe')
+    print _get_url_for_date_range('2009-10-02', '2009-10-04','ahead','1hr','O')
+    print _get_url_for_date_range('2009-10-02', '2009-10-04','behind','1day','H')
+    print _get_url_for_date_range(TimeRange('2012/06/01', '2012/06/05'),'ahead','1min','4He')
+    print _get_url_for_date_range(TimeRange('2008-06-01', '2010-06-05'),'behind','10min','Fe')
+    print _get_url_for_date_range(TimeRange('2012/06/01', '2012/06/05'),'ahead','1hr','O')
+    print _get_url_for_date_range(TimeRange('2012/06/01', '2012/06/05'),'behind','1day','H')
+    """
+
+
+
+    @staticmethod
+    def _parse_txt(filepath):
+        """
+        Parses a STEREO SIT file from
+        http://www.srl.caltech.edu/STEREO/Public/SIT_public.html
+        
+        and returns header and astropy.Table object containing data
+    
+        """
+        header = []
+        data = ascii.read(filepath, delimiter = "\s", data_start = 27) 
+    
+        # To read in column names
+        data_all = open(filepath)
+        for i, line in enumerate(data_all):
+            if i > 13 :
+                 header = header + [line]
+            if i > 23:
+                break
+        data_all.close()
```

Comment:
hmm I can also simply define the header list (rather than reading it in) for this instrument since all the files have same header? That'll be faster. Let me know your thoughts. In the meanwhile let me try reading it in with astropy only.